File: CCVTAC/CCVTAC.Console/PostProcessing/ImageProcessor.cs
using CCVTAC.Console.ExternalUtilities;

namespace CCVTAC.Console.PostProcessing;

internal static class ImageProcessor
{
    internal static void Run(string workingDirectory, Printer printer)
    {
        ExternalProgram imageProgram = new(
            "mogrify",
            "https://imagemagick.org/script/mogrify.php",
            "image cropping"
        );

        UtilitySettings imageEditToolSettings = new(
            imageProgram,
            "-trim -fuzz 10% *.jpg",
            workingDirectory
        );

        Runner.Run(imageEditToolSettings, printer);
    }
}

File: src/CCVTAC.FSharp/PostProcessing/ImageProcessor.fs
namespace CCVTAC.Console.PostProcessing

open CCVTAC.Console.ExternalTools

module ImageProcessor =

    let private programName = "mogrify"

    let run workingDirectory printer : unit =
        let toolSettings = workingDirectory |> ToolSettings.create $"{programName} -trim -fuzz 10%% *.jpg"
        Runner.runTool toolSettings [] printer |> ignore
